makeSpectral.py

```diff
@@ -0,0 +1,259 @@
+from os import listdir, rename
+from os.path import isfile, join, abspath, basename, splitext, dirname
+from typing import override
+from typing import TextIO, List
+from shutil import copy, SameFileError
+from argparse import ArgumentParser
+
+"""
+NOTE:
+	Although a little more cumbersome, I feel that providing the our
+	*replace substring* function extra verbosity is important.
+	This is because the User's chosen Name and Description
+	may occur in parts of the c++ code we do not want to edit.
+	The inclusion of the context around the needle we want to substitute
+	will help prevent accidental substitions.
+Happy Coding. Pix ❤️
+""" 
+
+class Cmd:
+	def __init__(self, p : str, n : str, d : str, v : bool):
+		self.path = p.pop() if p else None
+		self.name = n.pop() if n else None
+		self.description = d.pop() if d else None
+		self.verbose = v.pop() if v else False
+
+class Jucer:
+
+	def __init__(self, name:str, desc:str, path:str):
+		self.name = name
+		self.desc = desc
+		self.path = path
+		self.diry = dirname(path)
+
+
+	def swit(self, haystack:str, needle:str, spore:str) -> str :
+		if haystack == "" or haystack is None:
+			return None
+		if not needle in haystack:
+			raise ValueError("needle not found in haystack")
+		return haystack.replace(needle, spore, 1)
+
+	def manip(self, dt : str) -> str:
+		return dt
+
+	def rename_path(self, old_path : str) -> str:	
+		new_file_name = self.swit(basename(old_path), "Base", self.name)
+		new_file_path = join(self.diry, new_file_name)
+		rename(old_path, new_file_path)
+		return new_file_path
+
+	def crank(self) -> str:
+		print(self.path)
+		with open(self.path, 'r+') as f:
+			text = f.read()
+			manip_text = self.manip(text)
+			f.seek(0)
+			f.write(manip_text)
+			f.truncate()
+		self.path = self.rename_path(self.path)
+		return text
+
+class Base(Jucer):
+
+	def manip_description(self, dt:str) -> str:
+		dt = self.swit(dt, 'pluginDesc="BaseDescription"', f'pluginDesc="{self.desc}"')
+		dt = self.swit(dt, 'MAINGROUP id="NNl5AK" name="Base"', f'MAINGROUP id="NNl5AK" name="{self.name}"')
```

Comment:
I noticed in the committed NEWNAME.jucer file that the name had not changed. I think this is because the id is regenerated each time it's saved? Maybe we shouldn't rely on the ID being the same. 
Could just search for our name="Base" needle instead. Feel free to change the name in the base project file to make it easier to replace

Reply:
Ah good catch.
How about I change my needle from Base to "BaseTitle"
There is no other instance of `name="BaseTitle"` in the SpectralSuiteBuild/Base.jucer file.
I can then ditch all the additional context I only needed for accident prevention.

Reply:
No problem with that at all. Good idea